sunpy/map/tests/test_mapbase.py

```diff
@@ -135,8 +134,30 @@ def test_nickname_set(generic_map):
     assert generic_map.nickname == 'hi'
 
 
-def test_date(generic_map):
-    assert isinstance(generic_map.date, Time)
+date_dict = {'DATE-AVG': parse_time('2020-01-01'),
+             'DATE-OBS': parse_time('2020-02-01'),
+             'DATE-BEG': parse_time('2020-03-01'),
+             'DATE-END': parse_time('2020-03-03')}
+date_begend = date_dict['DATE-BEG'] + (date_dict['DATE-END'] - date_dict['DATE-BEG']) / 2
+
+
```

Comment:
What about the case where they are all missing?

Reply:
I've added a test for this.